File: Map.cpp
#include "Map.h"
#include <iostream>
#include <stdlib.h>

Map::Map(){
    this->cells.resize(10, vector<Cell*>(10));
};
Map::~Map(){
};

//setter

void Map::setCell(Cell* C, int x, int y){
    this->cells[x][y] = C;
}; //set sebuah cell di posisi x dan y

void Map::generateMap(vector<Cell>* vsea, vector<Cell>* vgrass){
    // mapnya gini kira kira:
    //
    // o o o o o o o o o o
    // o o o o o o o o o o
    // o o - - - - - - o o
    // o o - - - - - - o o
    // o o - - - - - - o o
    // o o - - - - - - o o
    // o o - - - - - - o o
    // o o - - - - - - o o
    // o o o o o o o o o o
    // o o o o o o o o o o
    //
    // blom fix, ganti gpp
    int vseaint = 0;
    int vgrassint = 0;
    for (int i = 0; i < this->size; i++)
    {
        for (int j = 0; j < this->size; j++)
        {
            if (i <= 2 || i >= 8 || j <= 2 || j >= 8){
                this->cells[i][j] = &vsea->at(vseaint);
                vseaint++;
            }
            else{
                this->cells[i][j] = &vgrass->at(vgrassint);
                vgrassint++;
            }
        }
        
    }
    
}; //Maksudnya adalah memasukkan cell ke cells, ini bisa didesain map nya gmn, atau random generated, belum aku putusin

//getter
Cell* Map::getCell(int x, int y){
    if (x>=0 || x<10 && (y>=0 && y<10))
        return cells[x][y];
    else
        return nullptr;
}; //get cell berdasarkan posisi

//function
void Map::printMap(Player* P){
    for (int i = 0; i < this->size; i++)
    {
        for (int j = 0; j < this->size; j++)
        {
            Cell* curr = this->cells[i][j];
            if (P->getPosX() == i && P->getPosY() == j){
                    curr->printSymbol(1);
            }
            else if (P->ActiveX() == i && P->ActiveY() == j){
                    curr->printSymbol(2);
            }
            else if (curr->isOccupied()){
                if (curr->getEnemy()->getPosX() == i && curr->getEnemy()->getPosY() == j){
                    curr->printSymbol(3);
                }
            }
            else{
                curr->printSymbol(4);
            }
        }
        cout<<"\n";
    }
    
};

File: java/Map.java
package willywangkyjava;

import java.util.List;
import java.io.*;

public class Map {
    final int size;
    Cell[][] cells;

    public Map() {
        this.size = 10;
        this.cells = new Cell[this.size][this.size];
    }

    public Map(int size) {
        this.size = size;
        this.cells = new Cell[this.size][this.size];
    }

    void setCell(Cell c, int x, int y) {
        this.cells[x][y] = c;
    }

    void generateMapFromFile() throws FileNotFoundException, IOException {
        File file = new File("Map.txt");
        FileReader fr = new FileReader(file);
        int c;
        int i = 0, j = 0;
        while ((c = fr.read()) != -1) {
            char ch = (char) c;
            if (ch == '\n') {
                i++;
                j = 0;
            } else if (ch == 'o') {
                this.cells[i][j] = new Cell("Sea");
                j++;
            } else if (ch == '-') {
                this.cells[i][j] = new Cell("Grassland");
            }
        }
    }

    void generateMap() {
        for (int i = 0; i < this.size; i++) {
            for (int j = 0; j < this.size; j++) {
                if (i <= 2 || i >= 8 || j <= 2 || j >= 8){
                    this.cells[i][j] = new Cell("Sea");
                }
                else{
                    this.cells[i][j] = new Cell("Grassland");
                }
            }
        }
    }

    Cell getCell(int x, int y) {
        if ((x >= 0 && x < this.size) && (y >= 0 || y < this.size)) {
            return this.cells[x][y];
        } else {
            return null;
        }
    }

    void printMap(Player P) {
        for (int i = 0; i < this.size; i++) {
            for (int j = 0; j < this.size; j++) {
                
                if (P.getPosX() == i && P.getPosY() == j){
                    this.cells[i][j].printSymbol(1);

                } else if (P.ActiveX() == i && P.ActiveY() == j) {
                    this.cells[i][j].printSymbol(2);

                } else if (this.cells[i][j].isOccupied()) {
                    if (this.cells[i][j].getEnemy().getPosX() == i && this.cells[i][j].getEnemy().getPosY() == j) {
                        this.cells[i][j].printSymbol(3);
                    }

                } else {
                    this.cells[i][j].printSymbol(4);
                }
            }
            System.out.println();
        }
    }
}
